### DISCUSSION TITLE
Admininistration functionality for [Simple|Direct]MessageListenerContainer via jmx

### DISCUSSION THREAD

**Discussion by rfelgent** [spring-projects/spring-amqp#1520] [Ideas]
Hi @all,

at my company we have created code that exposes basic admininistration functionality for `[Simple|Direct]MessageListenerContainer`: the so called `MessageListenerContainerAdmin`.

The current feature set is compromised of:

- start|stop of listener containers for queue x
- setting concurrency like maxConsumer, minConsumer for queue x
- setting batch like batchSize, receiveTimeout for queue x

The idea of a `MessageListenerContainerAdmin` is highly inspired by `AmqpAdmin`.
My company would like to contribute such code to the core spring-amqp framework.

If you people (especially @garyrussell as lead developer of spring-amqp) are interested in this feature, I could make a PR.

**Comment by garyrussell** [comment]
Sure! Sounds good; thanks. I will create an issue from this discussion.

**Reply by garyrussell** [reply]
https://github.com/spring-projects/spring-amqp/issues/1521
